app/connector/support/processor/src/main/java/io/syndesis/connector/support/processor/ErrorMapper.java

```diff
@@ -56,13 +58,45 @@ private ErrorMapper() {
     public static ErrorStatusInfo mapError(final Exception exception, final Map<String, String> errorResponseCodeMappings,
             final Integer responseCode) {
         Integer errorResponseCode = responseCode;
-        SyndesisConnectorException sce = SyndesisConnectorException.from(exception);
-        for (Map.Entry<String, String> mapping : errorResponseCodeMappings.entrySet()) {
-            if (mapping.getKey().matches(sce.getCategory())) {
-                errorResponseCode = Integer.valueOf(mapping.getValue());
-                break;
+        final ErrorStatusInfo info;
+        if (matchByException(exception, errorResponseCodeMappings)) {
+            info = new ErrorStatusInfo(
+                    Integer.valueOf(errorResponseCodeMappings.get(exception.getClass().getName())), 
+                    categoryFromClassName(exception.getClass().getSimpleName()),
+                    getMessage(exception));
+        } else {
+            SyndesisConnectorException sce = SyndesisConnectorException.from(exception);
+            if (matchByCategory(sce, errorResponseCodeMappings)) {
+                info = new ErrorStatusInfo(
+                        Integer.valueOf(errorResponseCodeMappings.get(sce.getCategory())), 
+                        sce.getCategory(),
+                        sce.getMessage());
+            } else {
+                info = new ErrorStatusInfo(errorResponseCode, sce.getCategory(), sce.getMessage());
             }
         }
-        return new ErrorStatusInfo(errorResponseCode, sce.getCategory(), sce.getMessage());
+        return info;
+ 
+    }
+
+    private static boolean matchByException(final Exception exception, final Map<String, String> errorResponseCodeMappings) {
+        return errorResponseCodeMappings.containsKey(exception.getClass().getName());
     }
-}
+
+    private static boolean matchByCategory(final SyndesisConnectorException sce, final Map<String, String> errorResponseCodeMappings) {
+        return errorResponseCodeMappings.containsKey(sce.getCategory());
+    }
+
+    private static String getMessage(final Exception exception) {
+        if (exception.getCause()!=null && exception.getCause().getMessage()!=null) {
+            return exception.getCause().getMessage();
+        } else {
+            return exception.getMessage();
+        }
+    }
+
+    private static String categoryFromClassName(final String exceptionClassName) {
+        return CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, exceptionClassName.replace("Exception","Error"));
```

Comment:
We could probably do some regexp magic instead.

Reply:
It's already in our dependency tree, so why not use it?

Reply:
agreed